x/evm/keeper/state_transition.go

```diff
@@ -122,6 +124,9 @@ func (k Keeper) VMConfig(ctx sdk.Context, msg core.Message, cfg *types.EVMConfig
 //  2. The requested height is from an previous height from the same chain epoch
 //  3. The requested height is from a height greater than the latest one
 func (k Keeper) GetHashFn(ctx sdk.Context) vm.GetHashFunc {
+	cache := make(map[int64]common.Hash)
```

Comment:
seems we never release the cache in any cases, maybe worth clean up after certain size?

Reply:
I just compared with https://github.com/ethereum/go-ethereum/blob/7d29fff415c9ad08ebb34e5052a89585ce60eee2/core/evm.go#L81
they also not clearing, but not sure that we should clear as leak here only on total count of blocks + N upcomming and only if smart contract has block.blockhash call, so should not be a problem. I could try to change from the map to a slice, should be less size.